source/index.ts

```diff
@@ -118,7 +118,7 @@ const isObject = (value: unknown): value is object => typeof value === 'object';
 
 is.undefined = isOfType<undefined>('undefined');
 is.string = isOfType<string>('string');
-is.number = isOfType<number>('number');
+is.number = (value: unknown): value is number => isOfType<number>('number')(value) && !is.nan(value);
```

Comment:
Extract isTypeOf(‘number’) to a constant before ‘is.number’.  No need to create it over and over